.travis.yml

```diff
@@ -1,58 +1,57 @@
-# For more information, see https://github.com/hvr/multi-ghc-travis
-language: c
+# For more information, see https://github.com/ChrisPenner/haskell-stack-travis-ci
+language: nix
 sudo: false
 
 git:
   submodules: false  # whether to recursively clone submodules
 
 cache:
   directories:
-    - $HOME/.cabal/packages
-    - $HOME/.cabal/store
-
-before_cache:
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/build-reports.log
-  # remove files that are regenerated by 'cabal update'
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/00-index.*
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/*.json
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.cache
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar.idx
+    - $HOME/.ghc
+    - $HOME/.cabal
+    - $HOME/.stack
+    - .stack-work
 
 matrix:
+  fast_finish: true
```

Comment:
I read the docs on this flag, but I don't quite understand how the behavior will change. Could you explain with an example?

Reply:
To be honest I haven't tried without the flag, I just copied from the [haskell-stack-travis-ci repo](https://github.com/ChrisPenner/haskell-stack-travis-ci). But the idea is to fail the build as soon as any job fails.

Reply:
Ok- that's fine with me. Either way, it's not a deal-breaker.